lib/Conversion/LWEToPolynomial/LWEToPolynomial.cpp

```diff
@@ -0,0 +1,126 @@
+#include "lib/Conversion/LWEToPolynomial/LWEToPolynomial.h"
+
+#include <cstddef>
+#include <optional>
+#include <utility>
+
+#include "lib/Conversion/Utils.h"
+#include "lib/Dialect/LWE/IR/LWEOps.h"
+#include "lib/Dialect/LWE/IR/LWETypes.h"
+#include "mlir/include/mlir/Dialect/Arith/IR/Arith.h"  // from @llvm-project
+#include "mlir/include/mlir/Dialect/Polynomial/IR/Polynomial.h"  // from @llvm-project
+#include "mlir/include/mlir/Dialect/Polynomial/IR/PolynomialAttributes.h"  // from @llvm-project
+#include "mlir/include/mlir/Dialect/Polynomial/IR/PolynomialOps.h"  // from @llvm-project
+#include "mlir/include/mlir/Dialect/Polynomial/IR/PolynomialTypes.h"  // from @llvm-project
+#include "mlir/include/mlir/Dialect/Tensor/IR/Tensor.h"  // from @llvm-project
+#include "mlir/include/mlir/IR/BuiltinOps.h"             // from @llvm-project
+#include "mlir/include/mlir/IR/BuiltinTypes.h"           // from @llvm-project
+#include "mlir/include/mlir/IR/ImplicitLocOpBuilder.h"   // from @llvm-project
+#include "mlir/include/mlir/IR/PatternMatch.h"           // from @llvm-project
+#include "mlir/include/mlir/IR/ValueRange.h"             // from @llvm-project
+#include "mlir/include/mlir/Support/LLVM.h"              // from @llvm-project
+#include "mlir/include/mlir/Support/LogicalResult.h"     // from @llvm-project
+#include "mlir/include/mlir/Transforms/DialectConversion.h"  // from @llvm-project
+
+namespace mlir::heir::lwe {
+
+#define GEN_PASS_DEF_LWETOPOLYNOMIAL
+#include "lib/Conversion/LWEToPolynomial/LWEToPolynomial.h.inc"
+
+class CiphertextTypeConverter : public TypeConverter {
+ public:
+  // Convert ciphertext to tensor<#dim x !poly.poly<#rings[#level]>>
+  // Our precondition (not verified) is that the key and plaintext have only one
+  // dimension, and the ciphertext has two dimensions. (i.e.) The key is a
+  // single polynomial and not a tensor of higher dimensions. We may support
+  // higher dimensions in the future (for schemes such as TFHE). Here are some
+  // TODO:
+  // 1. Add an encrypt/decrypt verifier to check for the preconditions of
```

Comment:
nit: update these comments

I think for (1) you could either op->walk at the start of the pass, or likely better, simply `return failure()` (with an `op.emitError() << expected 2 dimensional ciphertext`) if it doesn't match in the op rewrite pattern. You can file and attach an issue TODO on there to support higher dimensional rlwe encryption/decryption operations used in TFHE. But this way, the pass would fail because we added rlwe decrypt as an illegal op and the rewrite pattern fails to match.

Reply:
Ack.